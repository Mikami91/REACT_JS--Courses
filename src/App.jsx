import PropTypes from 'prop-types';

export function App({ title, subtitle }) {
  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      <h2 data-testid='test-subtitle'>{subtitle}</h2>
    </>
  );
}

App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

App.defaultProps = {
  // title: 'Dark-.-Ness',
};
