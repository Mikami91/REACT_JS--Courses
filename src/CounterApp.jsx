import { useState } from 'react';
import PropTypes from 'prop-types';

export function CounterApp({ title, subtitle, value }) {
  const [state, setState] = useState(value);

  const handleAdd = () => {
    setState(state + 1);
  };

  const handleLess = () => {
    setState(state > 0 ? state - 1 : 0);
  };

  const handleReset = () => {
    setState(value);
  };

  return (
    <>
      <h1>{title}</h1>
      <h2>{subtitle}</h2>
      <h3>{state}</h3>

      <button onClick={handleAdd}>Add</button>
      <button onClick={handleLess}>Less</button>
      <button id='btn-reset' aria-label='btn-reset' onClick={handleReset}>
        Reset
      </button>
    </>
  );
}

CounterApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  title: 'Counter App',
  subtitle: 'Value',
  value: 0,
};
