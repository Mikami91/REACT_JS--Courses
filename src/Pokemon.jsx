import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
// Slices
import { getPokemons } from './store/slices/pokemon';

const Pokemon = () => {
  const { isLoading, pokemons, page } = useSelector((state) => state.pokemons);
  const dispatch = useDispatch();

  console.log(page);

  useEffect(() => {
    dispatch(getPokemons());
  }, []);

  return (
    <>
      <h1>Pokemons</h1>
      <hr />
      <span>Loading: {isLoading ? 'true' : 'false'}</span>
      <ul>
        {pokemons.map((i) => (
          <li key={i.name}>{i.name}</li>
        ))}
      </ul>
      <button
        disabled={isLoading || page == 0}
        onClick={() => dispatch(getPokemons(page - 1))}
      >
        Prev
      </button>
      <button
        disabled={isLoading}
        onClick={() => dispatch(getPokemons(page + 1))}
      >
        Next
      </button>
    </>
  );
};

export default Pokemon;
