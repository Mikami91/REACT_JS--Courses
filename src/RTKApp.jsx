import { useState } from 'react';
import { useGetTodoQuery, useGetTodosQuery } from './store/apis';

const RTKApp = () => {
  const [todoId, setTodoId] = useState(1);
  // const { data: todos = [], isLoading } = useGetTodosQuery();
  const { data: todo, isLoading } = useGetTodoQuery(todoId);

  const prevTodo = () => setTodoId(todoId - 1);
  const nextTodo = () => setTodoId(todoId + 1);

  return (
    <>
      <h1>RTK Query</h1>
      <hr />
      <h4>Is Loading: {isLoading ? 'true' : 'false'}</h4>

      <pre>{JSON.stringify(todo)}</pre>
      <button onClick={prevTodo}>Prev</button>
      <button onClick={nextTodo}>Next</button>

      {/* <ul>
        {todos.map((i) => (
          <li key={i.id}>
            <strong> {i.completed ? 'DONE' : 'PENDING'} - </strong>
            {i.title}
          </li>
        ))}
      </ul> */}
      {/* <button>Next</button> */}
    </>
  );
};

export default RTKApp;
