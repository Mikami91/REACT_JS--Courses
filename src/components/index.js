export { default as GifGrid } from './GifGrid';
export { default as GifItem } from './GifItem';
export { default as Header } from './Header';
export { default as InputSearch } from './InputSearch';
