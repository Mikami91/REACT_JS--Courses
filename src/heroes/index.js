export * from './pages';
export * from './routers/HeroesRouter';
