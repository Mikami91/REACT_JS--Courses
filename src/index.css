body {
  margin: 0;
  padding: 1rem 2rem 2rem 2rem;
  min-height: 100vh;
  color-scheme: light dark;
  color: rgba(255, 255, 255, 0.87);
  background-color: #242424;
}

#mikami {
  color: #eb642f;
  font-family: 'Trebuchet MS';
  font-weight: bold;
  font-style: italic;
  font-size: large;
}

#header {
  background-color: #242424;
  height: 220px;
}

#footer {
  background-color: #242424;
  height: 50px;
}
