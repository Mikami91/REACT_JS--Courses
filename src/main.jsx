import React from 'react';
import ReactDOM from 'react-dom/client';
// App
import App from './App';
import Counter from './01-useState/Counter';
import CounterCustomHook from './01-useState/CounterCustomHook';
import Form from './02-useEffect/Form';
import FormCustomHook from './02-useEffect/FormCustomHook';
import MultiCustomHooks from './examples/MultiCustomHooks';
import FocusScreen from './03-useRef/FocusScreen';
import Layout from './04-useLayoutEffect/Layout';
import Memorize from './05-memo/Memorize';
import MemoHook from './06-useMemo/MemoHook';
import CallbackHook from './07-useCallback/CallbackHook';
import Father from './homework/Father';
// Styles
import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
  <Father />
  // </React.StrictMode>
);
