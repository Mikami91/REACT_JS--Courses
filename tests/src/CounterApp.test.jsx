import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../../src/CounterApp';

describe('TEST: CounterApp', () => {
  const initialValue = 0;

  test('Should make match with snapshot', () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test('Should show initial value: 100', () => {
    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy();

    expect(screen.getByRole('heading', { level: 3 }).innerHTML).toContain(
      '100'
    );
  });

  test('Should increase value with button Add', () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('Add'));
    expect(screen.getByText('1')).toBeTruthy();
  });

  test('Should decrease value with button Less', () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('Less'));
    expect(screen.getByText('0')).toBeTruthy();
  });

  test('Test button Reset', () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('Add'));
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

    expect(screen.getByText('0')).toBeTruthy();
  });
});
