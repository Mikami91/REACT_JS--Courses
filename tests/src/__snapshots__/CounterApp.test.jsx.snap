// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST: CounterApp Should make match with snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    Value
  </h2>
  <h3>
    0
  </h3>
  <button>
    Add
  </button>
  <button>
    Less
  </button>
  <button
    aria-label="btn-reset"
    id="btn-reset"
  >
    Reset
  </button>
</div>
`;
